crates/superposition/src/organisation/handlers.rs

```diff
@@ -0,0 +1,148 @@
+use super::types::{
+    CreateOrganisationRequest, CreateOrganisationResponse, OrganisationResponse,
+};
+use actix_web::{
+    get, post,
+    web::{self, Json, Query},
+    HttpResponse, Scope,
+};
+use chrono::Utc;
+use diesel::prelude::*;
+use idgenerator::IdInstance;
+use service_utils::service::types::DbConnection;
+use superposition_types::{
+    custom_query::PaginationParams, result as superposition, PaginatedResponse,
+};
+
+use superposition_types::database::{
+    models::organisation::Organisation, schema::organisation::dsl::organisation,
+};
+
+pub fn endpoints() -> Scope {
+    Scope::new("")
+        .service(create_organisation)
+        .service(list_organisations)
+        .service(get_organisation)
+}
+
+#[post("")]
+pub async fn create_organisation(
+    req: web::Json<CreateOrganisationRequest>,
+    db_conn: DbConnection,
+) -> superposition::Result<HttpResponse> {
+    let DbConnection(mut conn) = db_conn;
+    
+    // Generating a numeric ID from IdInstance and prefixing it with `orgid`
+    let numeric_id = IdInstance::next_id();
+    let org_id = format!("orgid{}", numeric_id);
+    let now = Utc::now().naive_utc();
+    
+    let new_org = Organisation {
+        id: org_id.clone(),
+        country_code: req.country_code.clone(),
+        contact_email: req.contact_email.clone(),
+        contact_phone: req.contact_phone.clone(),
+        created_by: req.created_by.clone(),
```

Comment:
this should not be a request param, this should be taken from the User's username